core/unix/coredump.c

```diff
@@ -479,26 +495,62 @@ os_dump_core_internal(dcontext_t *dcontext)
         return false;
     }
 
-    if (!write_elf_header(elf_file, /*entry_point=*/0,
+    if (!write_elf_header(elf_file, /*entry_point=*/get_entry_point(&mc),
                           /*section_header_table_offset*/ sizeof(ELF_HEADER_TYPE) +
-                              PROGRAM_HEADER_LENGTH + section_data_size,
+                              PROGRAM_HEADER_NOTE_LENGTH +
+                              sizeof(ELF_PROGRAM_HEADER_TYPE) * section_count +
+                              section_data_size,
                           /*flags=*/0,
-                          /*program_header_count=*/1,
+                          /*program_header_count=*/section_count,
                           /*section_header_count=*/section_count,
                           /*section_string_table_index=*/section_count - 1)) {
         os_close(elf_file);
         return false;
     }
+    // Write program header table entry.
     if (!write_program_header(elf_file, PT_NOTE, /*flags=*/0,
                               /*offset=*/sizeof(ELF_HEADER_TYPE) +
-                                  sizeof(ELF_PROGRAM_HEADER_TYPE),
+                                  sizeof(ELF_PROGRAM_HEADER_TYPE) * section_count,
```

Comment:
Should this be the program header count?  Clearest to make a new variable for it and use it above too.

Reply:
Yes. I've added a new variable.